File: samples/vite-react-api/apphost.py
import sys
from pathlib import Path
sys.path.insert(0, str(Path(__file__).parent / ".modules"))

from aspire import create_builder

builder = create_builder()

cache = builder.add_container("cache", "redis:latest")

api = builder.add_dockerfile("api", "./src/api")
api.with_environment("REDIS_HOST", "cache")
api.with_environment("REDIS_PORT", "6379")
api.with_http_endpoint(target_port=8080, env="PORT")
api.with_external_http_endpoints()

web = builder.add_dockerfile("web", "./src/web")
web.with_http_endpoint(target_port=5173, env="PORT")
web.with_external_http_endpoints()

app = builder.build()
app.run()

File: samples/vite-react-api/apphost.ts
// Aspire TypeScript AppHost for Vite + React + FastAPI + Redis
// For more information, see: https://aspire.dev

import { createBuilder, OtlpProtocol } from './.modules/aspire.js';

const builder = await createBuilder();

// Redis cache
const cache = builder.addContainer("cache", "redis:latest");

// FastAPI backend (./src/api has a Dockerfile) — OTel via gRPC (default).
const api = builder.addDockerfile("api", "./src/api")
  .withOtlpExporter()
  .withEnvironment("REDIS_HOST", "cache")
  .withEnvironment("REDIS_PORT", "6379")
  .withHttpEndpoint({ targetPort: 8080, env: "PORT" })
  .withExternalHttpEndpoints();

// Vite React frontend (./src/web has a Dockerfile).
//
// We OPT INTO HTTP/protobuf for OTel here so that the SAME endpoint URL the
// container uses for server-side telemetry can also be re-exposed to client
// code (the browser cannot speak gRPC). vite.config.ts re-exports the OTEL_*
// env vars as VITE_OTEL_* and rewrites host.docker.internal -> localhost so
// the running SPA can POST traces to the dashboard from the user's machine.
//
// The dashboard's HTTP/protobuf listener must be enabled before this works.
// `apphost.run.json` sets ASPIRE_DASHBOARD_OTLP_HTTP_ENDPOINT_URL on both
// launch profiles so plain `aspire run` works out of the box. See
// .squad/decisions/inbox/parker-vite-react-otel.md for the full design.
const web = builder.addDockerfile("web", "./src/web")
  .withOtlpExporter({ protocol: OtlpProtocol.HttpProtobuf })
  .withHttpEndpoint({ targetPort: 5173, env: "PORT" })
  .withExternalHttpEndpoints();

await builder.build().run();
